contracts/adapters/Managing.sol

```diff
@@ -98,6 +105,12 @@ contract ManagingContract is IManaging, DaoConstants, MemberGuard {
         require(_proposalId < type(uint64).max, "proposalId too big");
         uint64 proposalId = uint64(_proposalId);
         ProposalDetails memory proposal = proposals[_proposalId];
+				bytes32[] memory keys = proposal.keys;
```

Comment:
it makes sense to read those arrays later. It is only relevant AFTER the require.

So if one require fails, you save some gas by not reading the arrays you don't need

Reply:
Will fix.